bee-network/bee-autopeering/src/peering/manager.rs

```diff
@@ -0,0 +1,774 @@
+            }
+        } else {
+            log::debug!("Denying peering request: Peer filtered.");
+        }
+    } else {
+        log::debug!("Denying peering request: Peer not verified.");
+    }
+
+    // In any case send a response.
+    send_peering_response_to_addr(ctx.peer_addr, ctx.peer_id, ctx.msg_bytes, ctx.server_tx, status);
+}
+
+fn handle_peering_response<V: NeighborValidator>(
+    peer_res: PeeringResponse,
+    peer_reqval: RequestValue,
+    ctx: RecvContext,
+    nb_filter: &NeighborFilter<V>,
+) {
+    log::trace!("Handling peering response.");
+
+    let mut status = peer_res.status();
+
+    if status {
+        log::debug!("Peering accepted by {}.", ctx.peer_id);
+
+        let peer = ctx
+            .active_peers
+            .read()
+            .find(ctx.peer_id)
+            .cloned()
+            .expect("inconsistent peer list")
+            .into_peer();
+
+        // Hive.go: if the peer is already in inbound, do not add it and remove it from inbound
+        // TODO: investigate why!
+        if ctx.inbound_nbh.remove_neighbor(ctx.peer_id).is_some() {
+            // Change status to `false`.
+            status = false;
+
+            // Fire `OutgoingPeering` event with status = `false`.
+            publish_peering_event::<OUTGOING>(
+                peer.clone(),
+                status,
+                ctx.local,
+                ctx.event_tx,
+                ctx.inbound_nbh,
+                ctx.outbound_nbh,
+            );
+
+            // Drop that peer.
+            send_drop_peering_request_to_peer(peer, ctx.server_tx, ctx.event_tx, ctx.inbound_nbh, ctx.outbound_nbh);
+        } else if ctx.outbound_nbh.insert_neighbor(peer.clone(), ctx.local) {
+            // Update the neighbor filter.
+            nb_filter.add(*peer.peer_id());
+
+            // Fire `OutgoingPeering` event with status = `true`.
+            publish_peering_event::<OUTGOING>(peer, status, ctx.local, ctx.event_tx, ctx.inbound_nbh, ctx.outbound_nbh);
+        } else {
+            log::debug!("Failed to add neighbor to outbound neighborhood after successful peering request");
+        }
+    } else {
+        log::debug!("Peering by {} denied.", ctx.peer_id);
+    }
+
+    // Send the response notification.
+    if let Some(tx) = peer_reqval.response_tx {
+        tx.send(ctx.msg_bytes.to_vec()).expect("error sending response signal");
+    }
+}
+
+fn handle_drop_request<V: NeighborValidator>(
+    _drop_req: DropPeeringRequest,
+    ctx: RecvContext,
+    nb_filter: &NeighborFilter<V>,
+) {
+    log::trace!("Handling drop request.");
+
+    let mut removed_nb = ctx.inbound_nbh.remove_neighbor(ctx.peer_id);
+
+    if let Some(nb) = ctx.outbound_nbh.remove_neighbor(ctx.peer_id) {
+        removed_nb.replace(nb);
+
+        nb_filter.add(*ctx.peer_id);
+
+        // TODO: trigger immediate outbound neighborhood update; currently we wait for the next interval
```

Comment:
Just in case

Reply:
I'ld like to move that to a future patch version as it's not a critical thing.